py4DSTEM/process/latticevectors/index.py

```diff
@@ -53,3 +53,158 @@ def index_bragg_directions(x0, y0, ux, uy, vx, vy, bragg_x, bragg_y):
+        y -= y0
+        x -= x0
+        h_max = np.max(np.ceil(np.abs((x/ux,y/uy))))
+        k_max = np.max(np.ceil(np.abs((x/vx,y/vy))))
+    
+    (hlist,klist) = np.meshgrid(np.arange(-h_max,h_max+1),np.arange(-k_max,k_max+1))
+
+    M_ideal = np.vstack((hlist.ravel(),klist.ravel())).T
+    ideal_peaks = np.matmul(M_ideal,beta)
+
+    ideal_data = np.core.records.fromarrays(np.array([ideal_peaks[:,0],ideal_peaks[:,1],M_ideal[:,0],M_ideal[:,1]]),\
+                                        names='qx,qy,h,k',formats='float,float,int,int')
+
+    coords = [('qx',float),('qy',float),('h',int),('k',int)]
+    ideal_lattice = PointList(coordinates=coords)
+    ideal_lattice.add_dataarray(ideal_data)
+
+    #shift to the DP center
+    ideal_lattice.data['qx'] += x0
+    ideal_lattice.data['qy'] += y0
+
+    # trim peaks outside the image
+    deletePeaks = (ideal_lattice.data['qx'] > Q_Nx) | \
+            (ideal_lattice.data['qx'] < 0) | \
+            (ideal_lattice.data['qy'] > Q_Ny) | \
+            (ideal_lattice.data['qy'] < 0)
+    ideal_lattice.remove_points(deletePeaks)
+
+    return ideal_lattice
+
+def index_point_list_array(pointlistarray, lattice, maxPeakSpacing, mask=None):
+    """
+    Indexes the Bragg peaks in a PointListArray based on the indexed points in
+    lattice, which are placed in 'h' and 'k' coordinates in the PointLists in the array.
+    If `mask` is specified, only the locations where mask is True are indexed.
+
+    Accepts:
+        pointlistarray          (PointListArray) the braggpeaks pointlist to index
+        lattice           (PointList) a PointList containing with indexed peaks,
+                                    with the four coordinates 'qx', 'qy', 'h', and 'k'
+                                    for the lattice points and corresponding indices
+        maxPeakSpacing          (float) Maximum distance from the ideal lattice points
+                                    to include a peak for indexing
+        mask                    (bool)  Boolean mask, same shape as the pointlistarray,
+                                    indicating which locations should be indexed. This
+                                    can be used to index different regions of the scan
+                                    with different lattices
+
+    Returns:
+        pointlistarray          The original pointlistarray, with new coordinates 'h' and 'k'
+                                    containing the indices of each indexable peak 
+    """
+
+    assert isinstance(pointlistarray,PointListArray)
+    assert np.all([name in pointlistarray.dtype.names for name in ('qx','qy','intensity')])
+    assert isinstance(lattice, PointList)
+    assert np.all([name in lattice.dtype.names for name in ('qx','qy','h','k')])
+
+    if mask is None:
+        mask = np.ones(pointlistarray.shape,dtype=bool)
+
+    assert np.all(mask.shape == pointlistarray.shape), 'mask must have same shape as pointlistarray'
```

Comment:
Remove np.all (mask.shape and pointlistarray.shape are tuples, so the == statement will always be just True or False)

Reply:
👍